src/main/java/games/strategy/engine/framework/map/download/DownloadCoordinator.java

```diff
@@ -1,107 +1,147 @@
 package games.strategy.engine.framework.map.download;
 
-import java.util.HashMap;
+import java.util.Collection;
 import java.util.HashSet;
-import java.util.Map;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Queue;
 import java.util.Set;
-import java.util.function.Consumer;
-import java.util.function.Predicate;
+import java.util.concurrent.CopyOnWriteArrayList;
 
-import games.strategy.debug.ClientLogger;
 import games.strategy.engine.ClientFileSystemHelper;
-import games.strategy.util.ThreadUtil;
 
 /**
  * Class that accepts and queues download requests. Download requests are started in background
  * thread, this class ensures N are in progress until all are done.
  */
-public class DownloadCoordinator {
+final class DownloadCoordinator {
+  static final DownloadCoordinator INSTANCE = new DownloadCoordinator();
```

Comment:
Do you think it would be more appropriate to move this singleton instance to `ClientContext`?  It will unfortunately require this class and its constructor to move up to public visibility, but it seems like this was the purpose `ClientContext` was created for.

Alternatively, one of the UI classes in this package could host the instance.  It looks like only `DownloadMapsWindow` and `MapDownloadProgressPanel` reference it.  And since `DownloadMapsWindow` is responsible for creating `MapDownloadProgressPanel`, it could easily pass the reference downstream.  This is probably the cleaner solution because it keeps everything within this package.  What do you think?